src/main/java/org/opensearch/security/privileges/PrivilegesEvaluator.java

```diff
@@ -304,7 +309,17 @@ public PrivilegesEvaluationContext createContext(
         TransportAddress caller = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_REMOTE_ADDRESS);
         ImmutableSet<String> mappedRoles = ImmutableSet.copyOf((injectedRoles == null) ? mapRoles(user, caller) : injectedRoles);
 
-        return new PrivilegesEvaluationContext(user, mappedRoles, action0, request, task, irr, resolver, clusterStateSupplier);
+        return new PrivilegesEvaluationContext(
```

Comment:
I think this should be split into 2 separate classes: `RoleBasedPrivilegesEvaluatorContext` vs `PermissionBasedPrivilegesEvaluatorContext` instead of trying to overload this class to support 2 separate use-cases.

Reply:
Done - this is cleaner IMO